optimum/intel/ipex/modeling_base.py

```diff
@@ -106,9 +108,9 @@ def __init__(
 
         # Non-generation tasks can use torch.compile to get acceleration.
         if (
-            model.device.type == "cpu"
-            and self.export_feature not in _IPEX_EXPORTED_GENERATION_TASKS
-            and config.model_type not in _COMPILE_NOT_READY_MODEL_TYPES
+            self.model.device.type == "cpu"
+            and self.export_feature not in _IPEX_SUPPORTED_GENERATION_TASKS
```

Comment:
Hi @IlyasMoutawwakil . You can see we excluded the text-generation tasks when calling torch.compile here.

Reply:
hmm I see, can you give some explanation as to why it's not supported ? I believe many CausalLM models in transformers are now `torch.compile` compatible when static cache is in use. The use `_IPEX_SUPPORTED_GENERATION_TASKS` is confusing here, as it states "supported", maybe we can have a variable for `_IPEX_NOT_SUPPORTED_TORCH_COMPILE_TASKS` for example.

Also I see that for text2text, torch.compile is applied in the `__init__` and then skipped here, which is very confusing readability wise, it makes more sense to have an `apply_torch_compile` method that is overridden in IPEXSeq2Seq (for example to warmup with generate) removing `text2text` from "`_IPEX_NOT_SUPPORTED_TORCH_COMPILE_TASKS`"

Reply:
Hi @IlyasMoutawwakil . Sorry for not making it clear when I said text-generation is not compatible with torch.compile, it's because we patched a lot of ipex ops in text-generation models (such as llama; but compile works well for non-patching models as mistral). These patching ops will cause performance regression and we are fixing it, but need some time. I will indicate it and enable compile for non-patching models in text-generation task.